marlin-cli/commands/install.py

```diff
@@ -1,10 +1,126 @@
 import click
+import json
+import sys
+import os
+import subprocess
+import tarfile
+import requests
+import shutil
+from api import modules, archetypes
+from util import yaml_merge
+from util import json_merge
 
+def update_project(module_details, project_root): 
+    project_files = module_details.get("project_updates")
+    for source, destination in project_files:
+        project_dest = os.path.join(project_root, destination)
+        _, file_extension = os.path.splitext(source)
+        match file_extension:
+            case '.json': 
+                json_merge(source, project_dest)
+            case '.yaml' | '.yml': 
+                yaml_merge(source, project_dest)
+            case other: 
+                click.echo(click.style(f'Merge unsupported for file extension {other}', fg='red'))
+                sys.exit(1)
+
+def copy_source(module_details, project_root):
+    source_targets = module_details.get("raw_source_code")
```

Comment:
so this would then be the location in the repo that has the raw source code?

Reply:
This would be a map of module paths from root to client paths from root